src/util/address.rs

```diff
@@ -177,49 +177,47 @@ impl FromStr for AddressType {
     }
 }
 
-/// Version of the WitnessProgram: first byte of `scriptPubkey` in
-/// transaction output for transactions starting with opcodes ranging from 0
-/// to 16 (inclusive).
+/// Version of the witness program: first byte of `scriptPubkey` in transaction output for
+/// transactions starting with opcodes ranging from 0 to 16 (inclusive).
 ///
-/// Structure helps to limit possible version of the witness according to the
-/// specification; if a plain `u8` type was be used instead it would mean that
-/// version may be > 16, which is incorrect.
+/// Helps to limit possible version of the witness according to the specification; if a plain `u8`
```

Comment:
Is this proper English grammar? I'm not a native but I think it should be "Helps limit possible versions".

Also, I think it'd be nice to replace semicolon with full stop and make the rest a new paragraph so that shorter summary is displayed in the docs.

Reply:
I'm a native speaker but here in Australia our education of grammar is quite poor, I've done my best over the years to try and teach myself but I'm a long way from being an expert. I do enjoy looking these things up when I come across them though. In this case (I didn't look this up :) I think the sentence should probably have a 'the' in it, but we programmers often leave the 'the' out. If it reads "Helps to limit the possible version number of the witness ..." I believe it is correct because then it is clear that its singular i.e., the discreet possible version. Your sentence is correct however and I rekon it is better, will use as suggested.

> Also, I think it'd be nice to replace semicolon with full stop and make the rest a new paragraph so that shorter summary is displayed in the docs.

100% will do as suggested.

I'll re-spin tomorrow, thanks.

Reply:
Yeah, I forgot "the" - my biggest flaw in English is forgetting them. :)

Reply:
Its so common in technical writing that I often leave them out. Any time I get my writing proof read by non-technical folk they always pull me up on it.